src/faction_camp.cpp

```diff
@@ -1420,20 +1408,136 @@ void basecamp::craft_construction( npc &p, const std::string &cur_id, const std:
     }
 }
 
-bool talk_function::camp_garage_chop_start( npc &p, const std::string &task )
+/*
+ * All three camp_farm_ functions use the same logic, but passing the npc *comp to this function
+ * doesn't work for reasons I don't understand.  So repeat this stuff in two places: once
+ * farm_description and farm_start, and once for farm_return.
+ */
+static size_t camp_farm_act( const tripoint &omt_tgt, std::string &crops, farm_ops op )
 {
-    std::string dir = camp_direction( task );
-    const tripoint omt_pos = p.global_omt_location();
-    tripoint omt_trg;
-    std::vector<std::pair<std::string, tripoint>> om_expansions = om_building_region( p, 1, true );
-    for( const auto &e : om_expansions ) {
-        if( dir == om_simple_dir( omt_pos, e.second ) ) {
-            omt_trg = e.second;
+    size_t harvestable = 0;
+    size_t plots_empty = 0;
+    size_t plots_plow = 0;
+    bool plow = op & farm_ops::plow;
+    bool plant = op & farm_ops::plant;
+    bool harvest = op & farm_ops::harvest;
+
+    const auto is_dirtmound = []( const tripoint & pos, tinymap & bay1, tinymap & bay2 ) {
+        const ter_str_id &dirtmound = ter_str_id( "t_dirtmound" );
+        const furn_str_id &f_null = furn_str_id( "f_null" );
+        return ( bay1.ter( pos ) == dirtmound ) && ( bay2.furn( pos ) == f_null );
+    };
+    const auto is_unplowed = []( const tripoint & pos, tinymap & bay ) {
+        return bay.ter( pos ) == ter_str_id( "t_dirt" ) || bay.ter( pos ) == ter_str_id( "t_grass" );
+    };
+    const auto is_valid_seed = []( const item & itm ) {
+        return itm.is_seed() && itm.typeId() != "marloss_seed" && itm.typeId() != "fungal_seeds";
+    };
+
+    std::vector<std::string> plant_names;
+
+    //bay_json is what the are should look like according to jsons
+    tinymap bay_json;
+    bay_json.generate( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, calendar::turn );
+    //bay is what the area actually looks like
+    tinymap bay;
+    bay.load( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, false );
+    tripoint mapmin = tripoint( 0, 0, omt_tgt.z );
+    tripoint mapmax = tripoint( 2 * SEEX - 1, 2 * SEEY - 1, omt_tgt.z );
+    for( const tripoint &pos : bay.points_in_rectangle( mapmin, mapmax ) ) {
+        //Needs to be plowed to match json
+        if( is_dirtmound( pos, bay_json, bay ) && is_unplowed( pos, bay ) ) {
+            plots_plow += 1;
+        }
+        if( is_dirtmound( pos, bay, bay ) ) {
+            plots_empty++;
+        }
+        if( bay.furn( pos ) == furn_str_id( "f_plant_harvest" ) && !bay.i_at( pos ).empty() ) {
+            const item &seed = bay.i_at( pos ).front();
+            if( is_valid_seed( seed ) ) {
+                harvestable++;
+                const islot_seed &seed_data = *seed.type->seed;
+                item tmp = item( seed_data.fruit_id, calendar::turn );
+                if( std::find( plant_names.begin(), plant_names.end(),
+                               tmp.type_name( 3 ) ) != plant_names.end() ) {
```

Comment:
You can use `item::nname` which only requires a `itype_id` (and optionally a count) so you don't need to construct a dummy item.